display/d.mon/list.c

```diff
@@ -136,6 +139,9 @@ void list_files(const char *name, FILE *fd_out)
     strcat(tmpdir, "MONITORS");
     strcat(tmpdir, "/");
     strcat(tmpdir, name);
+    if (strlen(tmpdir) >= GPATH_MAX) {
+        G_fatal_error(_("The path for monitor <%s> is too long."), name);
+    }
```

Comment:
This type of check actually does not work. The point is not to limit the length of the string just because. The point is to not overflow the allocated buffer. You need to compute the length before using strcat, for strings which are too long, the behavior is undefined ([see strcat doc](https://en.cppreference.com/w/c/string/byte/strcat)).

Reply:
same usage of variable available_space for calculating the buffer availability